crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -14,25 +14,21 @@ use reth_db::{
 use reth_interfaces::provider::{ProviderError, ProviderResult};
 use reth_nippy_jar::NippyJar;
 use reth_primitives::{
-    snapshot::HighestSnapshots, Address, Block, BlockHash, BlockHashOrNumber, BlockNumber,
-    BlockWithSenders, ChainInfo, Header, Receipt, SealedBlock, SealedBlockWithSenders,
-    SealedHeader, SnapshotSegment, TransactionMeta, TransactionSigned, TransactionSignedNoHash,
-    TxHash, TxNumber, Withdrawal, B256, U256,
+    snapshot::{find_fixed_range, HighestSnapshots, SegmentHeader},
+    Address, Block, BlockHash, BlockHashOrNumber, BlockNumber, BlockWithSenders, ChainInfo, Header,
+    Receipt, SealedBlock, SealedBlockWithSenders, SealedHeader, SnapshotSegment, TransactionMeta,
+    TransactionSigned, TransactionSignedNoHash, TxHash, TxNumber, Withdrawal, B256, U256,
 };
 use std::{
     collections::{hash_map::Entry, BTreeMap, HashMap},
     ops::{Range, RangeBounds, RangeInclusive},
     path::{Path, PathBuf},
+    sync::Arc,
 };
 use tokio::sync::watch;
 
-/// Alias type for a map that can be queried for transaction/block ranges from a block/transaction
-/// segment respectively. It uses `BlockNumber` to represent the block end of a snapshot range or
-/// `TxNumber` to represent the transaction end of a snapshot range.
-///
-/// Can be in one of the two formats:
-/// - `HashMap<SnapshotSegment, BTreeMap<BlockNumber, RangeInclusive<TxNumber>>>`
-/// - `HashMap<SnapshotSegment, BTreeMap<TxNumber, RangeInclusive<BlockNumber>>>`
+/// Alias type for a map that can be queried for block ranges from a transaction
+/// segment respectively. It uses `TxNumber` to represent the transaction end of a snapshot range.
 type SegmentRanges = HashMap<SnapshotSegment, BTreeMap<u64, RangeInclusive<u64>>>;
```

Comment:
is this actually
```suggestion
type SegmentRanges = HashMap<SnapshotSegment, BTreeMap<BlockNumber, RangeInclusive<u64>>>;
```
?

where second `u64` can be either `TxNumber` or `BlockNumber`?

Reply:
`type SegmentRanges = HashMap<SnapshotSegment, BTreeMap<TxNumber, RangeInclusive<BlockNumber>>>;`

added the types, but currently it's only being used like this.